tests/e2e/e2e_test.go

```diff
@@ -111,3 +109,257 @@ func (s *IntegrationTestSuite) TestBankTokenTransfer() {
+	s.T().Logf("Depositing Gov Proposal: Software Upgrade")
+	s.depositGovProposal(chainAAPIEndpoint, sender, proposalCounter)
+	s.T().Logf("Weighted Voting Gov Proposal: Software Upgrade")
+	s.voteGovProposal(chainAAPIEndpoint, sender, proposalCounter, "yes=0.8,no=0.1,abstain=0.05,no_with_veto=0.05", true)
+
+	s.verifyChainHaltedAtUpgradeHeight(s.chainA, 0, proposalHeight)
+	s.T().Logf("Successfully halted chain at height %d", proposalHeight)
+
+	currentChain := s.chainA
+
+	for valIdx := range currentChain.validators {
+		var opts docker.RemoveContainerOptions
+		opts.ID = s.valResources[currentChain.id][valIdx].Container.ID
+		opts.Force = true
+		s.dkrPool.Client.RemoveContainer(opts)
+		s.T().Logf("Removed Container: %s", s.valResources[currentChain.id][valIdx].Container.Name[1:])
+	}
+
+	s.T().Logf("Restarting containers")
+	s.SetupSuite()
+
+	s.Require().Eventually(
+		func() bool {
+			h := s.getLatestBlockHeight(s.chainA, 0)
+			s.Require().NoError(err)
+
+			return (h > 0)
+		},
+		30*time.Second,
+		5*time.Second,
+	)
+
+	proposalCounter = 0
+}
+
+func (s *IntegrationTestSuite) TestGovCancelSoftwareUpgrade() {
+	chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+	senderAddress, err := s.chainA.validators[0].keyInfo.GetAddress()
+	s.Require().NoError(err)
+	sender := senderAddress.String()
+	height := s.getLatestBlockHeight(s.chainA, 0)
+	proposalHeight := height + 50
+	proposalCounter++
+
+	s.T().Logf("Writing proposal %d on chain %s", proposalCounter, s.chainA.id)
+	s.writeGovUpgradeSoftwareProposal(s.chainA, proposalHeight)
+
+	s.T().Logf("Submitting Gov Proposal: Software Upgrade")
+	s.submitNewGovProposal(chainAAPIEndpoint, sender, proposalCounter, "/root/.gaia/config/proposal_3.json")
+	s.depositGovProposal(chainAAPIEndpoint, sender, proposalCounter)
+	s.voteGovProposal(chainAAPIEndpoint, sender, proposalCounter, "yes", false)
+
+	proposalCounter++
+
+	s.T().Logf("Submitting Gov Proposal: Cancel Software Upgrade")
+	s.submitNewGovProposal(chainAAPIEndpoint, sender, proposalCounter, "/root/.gaia/config/proposal_4.json")
+	s.depositGovProposal(chainAAPIEndpoint, sender, proposalCounter)
+	s.voteGovProposal(chainAAPIEndpoint, sender, proposalCounter, "yes", false)
+
+	s.verifyChainPassesUpgradeHeight(s.chainA, 0, proposalHeight)
+	s.T().Logf("Successfully canceled upgrade at height %d", proposalHeight)
+}
+
+func (s *IntegrationTestSuite) fundCommunityPool(chainAAPIEndpoint string, sender string) {
+	s.Run("fund_community_pool", func() {
+		s.execDistributionFundCommunityPool(s.chainA, 0, chainAAPIEndpoint, sender, tokenAmount.String(), fees.String())
+
+		s.Require().Eventually(
+			func() bool {
+				afterDistPhotonBalance, err := getSpecificBalance(chainAAPIEndpoint, distModuleAddress, tokenAmount.Denom)
+				if err != nil {
+					s.T().Logf("Error getting balance: %s", afterDistPhotonBalance)
+				}
+				s.Require().NoError(err)
+
+				return afterDistPhotonBalance.IsValid()
```

Comment:
shouldn't the balance be previous balance + tokenAmount to confirm it was received?
just checking isValid means that pool has any balance right?

Reply:
So the issue is if the test is run in isolation or first, it will have no initial balance. We could check for that case and set it to 0 in the test. We can never assume it will always be `0photon` as the community pool will show a different balance depending on how many tests were run before. See:

Run in isolation
```
=== RUN   TestIntegrationTestSuite/TestSendTokensFromNewGovAccount/fund_community_pool
    e2e_test.go:236: Token Amount: 3300000000photon
    e2e_util_test.go:181: Executing gaiad tx distribution fund-community-pool on chain chain-zeafQW
    e2e_util_test.go:198: Successfully funded community pool
    e2e_test.go:248: After Balance: 3300330000photon
    e2e_test.go:251: Send Amount: 3300000000photon
    e2e_test.go:252: Difference: 330000photon
        --- PASS: TestIntegrationTestSuite/TestSendTokensFromNewGovAccount/fund_community_pool (11.18s)
```

Run after several gov tests
```
=== RUN   TestIntegrationTestSuite/TestSendTokensFromNewGovAccount/fund_community_pool
    e2e_test.go:237: Token Amount: 3300000000photon
    e2e_util_test.go:181: Executing gaiad tx distribution fund-community-pool on chain chain-8jbkQ3
    e2e_util_test.go:198: Successfully funded community pool
    e2e_test.go:249: After Balance: 399300660000photon
    e2e_test.go:252: Send Amount: 3300000000photon
    e2e_test.go:253: Difference: 396000660000photon
```